core/tauri/src/scope/fs.rs

```diff
@@ -258,6 +271,32 @@ impl Scope {
       false
     }
   }
+
+  /// This removes any paths added to allowed and forbidden scope by resetting
+  /// to the scope state to the given FsAllowlistScope config
+  pub fn reset<R: crate::Runtime>(
+    &self,
+    app: &App<R>,
+    scope: &FsAllowlistScope,
```

Comment:
instead of accepting the default scope as an arugment, we should probably store the default (the one passed to `for_fs_api`) on the `Scope` struct and reuse it here

Reply:
@amrbashir I wrote the changes but before I push I wanted to ask your opinion on this:

What if we only had one function `reset` that takes no arguments? We can store the initial allowed/forbidden patterns whenever Scope is created, similar to how you described it. The reset function will just restore allowed/forbidden scope back to it's original state.

```rs
pub fn reset(&self) -> crate::Result<()>
```

My reasoning:
1. If the app describes a set of forbidden directories in configuration, it probably shouldn't be cleared (the developer shouldn't include it otherwise). And I am thinking the same for any allowed paths
2. For version 2, I believe scopes are created with a manager, so the reset signature would change from accepting an `App<R>` to `Manager<R>` (probably not a big deal since App implements Manager)

Reply:
> ```rs
> pub fn reset(&self) -> crate::Result<()>
> ```
yeah this sounds better